entgql/schema.go

```diff
@@ -478,6 +488,128 @@ func (e *schemaGenerator) buildWhereInput(t *gen.Type, gqlType string) (*ast.Def
+		}
+		for _, f := range fields {
+			ant, err := annotation(f.Annotations)
+			if err != nil {
+				return nil, err
+			}
+			if ant.Skip.Is(SkipMutationCreateInput) || f.IsEdgeField() {
+				continue
+			}
+
+			scalar := e.mapScalar(f, ant, inputObjectFilter)
+			if scalar == "" {
+				return nil, fmt.Errorf("%s is not supported as input for %s", f.Name, name)
+			}
+			def.Fields = append(def.Fields, &ast.FieldDefinition{
+				Name:        camel(f.Name),
+				Type:        namedType(scalar, f.Optional),
+				Description: f.Comment(),
+			})
+		}
+		for _, e := range t.Edges {
+			ant, err := annotation(e.Annotations)
+			if err != nil {
+				return nil, err
+			}
+			if ant.Skip.Is(SkipMutationCreateInput) {
+				continue
+			}
+
+			if e.Unique {
+				def.Fields = append(def.Fields, &ast.FieldDefinition{
+					Name: camel(e.Name + "_ID"),
+					Type: namedType("ID", e.Optional),
+				})
+			} else {
+				def.Fields = append(def.Fields, &ast.FieldDefinition{
+					Name: camel(singular(e.Name) + "_IDs"),
+					Type: namedType("[ID!]", e.Optional),
+				})
+			}
+		}
+
+		defs = append(defs, def)
+	}
+	if !ant.Skip.Is(SkipMutationUpdateInput) {
+		fields := t.MutableFields()
+
+		name := fmt.Sprintf("Update%sInput", gqlType)
+		def := &ast.Definition{
+			Name:        name,
+			Kind:        ast.InputObject,
+			Description: fmt.Sprintf("%s is used for update %s object.\nInput was generated by ent.", name, gqlType),
+		}
+		for _, f := range fields {
+			ant, err := annotation(f.Annotations)
+			if err != nil {
+				return nil, err
+			}
+			if ant.Skip.Is(SkipMutationUpdateInput) {
+				continue
+			}
+
+			scalar := e.mapScalar(f, ant, inputObjectFilter)
+			if scalar == "" {
+				return nil, fmt.Errorf("%s is not supported as input for %s", f.Name, name)
+			}
+			def.Fields = append(def.Fields, &ast.FieldDefinition{
+				Name:        camel(f.Name),
+				Type:        namedType(scalar, true),
+				Description: f.Comment(),
+			})
+			if f.Optional {
+				def.Fields = append(def.Fields, &ast.FieldDefinition{
+					Name: camel("clear_" + f.Name),
+					Type: namedType("Boolean", true),
+				})
+			}
+		}
+		for _, e := range t.Edges {
+			ant, err := annotation(e.Annotations)
+			if err != nil {
+				return nil, err
+			}
+			if ant.Skip.Is(SkipMutationCreateInput) {
+				continue
+			}
+
+			if e.Unique {
+				def.Fields = append(def.Fields, &ast.FieldDefinition{
+					Name: camel(e.Name + "_ID"),
+					Type: namedType("ID", true),
+				}, &ast.FieldDefinition{
+					Name: camel(snake(e.MutationClear())),
+					Type: namedType("Boolean", true),
+				})
+			} else {
+				def.Fields = append(def.Fields, &ast.FieldDefinition{
+					Name: camel("add_" + singular(e.Name) + "_IDs"),
+					Type: namedType("[ID!]", true),
+				}, &ast.FieldDefinition{
+					Name: camel("remove_" + singular(e.Name) + "_IDs"),
+					Type: namedType("[ID!]", true),
+				})
```

Comment:
Due to a "bug" with "IDs", I can't use `MutationAdd()` and `MutationRemove()` functions

`camel(snake(e.MutationRemove()))` return `removeGroupIds`
and `camel(e.MutationRemove())` return `removegroupids`

So, I have to change copy it logic: `camel("remove_" + singular(e.Name) + "_IDs")`